dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -2684,8 +2680,8 @@ Spdp::SpdpTransport::write_i(WriteFlags flags)
   wbuff_.reset();
   CORBA::UShort options = 0;
   DCPS::Serializer ser(&wbuff_, encoding_plain_native);
-  if (!(ser << hdr_) || !(ser << data_) || !(ser << encap_LE) || !(ser << options)
-      || !(ser << plist)) {
+  DCPS::EncapsulationHeader encap(encoding_plain_native, DCPS::MUTABLE);
```

Comment:
Also @mitza-oci and @simpsont-oci this was saying little endian before while actually being native endianness. Does it actually have to be little endian or is native okay?

Reply:
The EncapsulationHeader has an explicit serialization function pair that explicitly orders the kind and option values into a byte array before serializing via an octet array operation. Therefore the manipulation of the byte-swapping is no longer needed.